File: Orquestra.Domain/Entities/Quote.cs
using Orquestra.Domain.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Orquestra.Domain.Entities;

public sealed class Quote : Audit
{
    [Key]
    public Guid QuoteId { get; set; }

    public Guid CompanyId { get; set; }
    [ForeignKey(nameof(CompanyId))]
    public Company? Company { get; set; }

    public Guid ClientId { get; set; }
    [ForeignKey(nameof(ClientId))]
    public Client? Client { get; set; }

    [MaxLength(120)]
    public string? Title { get; set; }

    [MaxLength(255)]
    public string? Observation { get; set; }

    public DateTime? ValidUntil { get; set; }

    public QuoteStatusEnum QuoteStatus { get; set; } = QuoteStatusEnum.Draft;

    public List<QuoteItem> Items { get; set; } = [];
}

File: Orquestra.Frontend/app/api/consts/quote.ts
import { BASE } from '@/app/api/fetch';
import { Guid } from 'guid-typescript';
import { iClient } from './client';
import { iCompanyOutput } from './company';

const controller = 'api/Quote';

export const CONSTS_QUOTE = {
    post: `${BASE}/${controller}`,
    put: `${BASE}/${controller}`,
    get: `${BASE}/${controller}`,
    getAllByCompanyId: `${BASE}/${controller}/GetAllByCompanyId`
};

export interface iQuote {
    quoteId?: Guid;
    companyId?: Guid;
    company?: iCompanyOutput | null;
    clientId?: Guid;
    client?: iClient | null;
    title?: string | null;
    observation?: string | null;
    validUntil?: Date | null;
    quoteStatus?: string;
    items?: iQuoteItem[];
}

export interface iQuoteItem {
    quoteItemId?: Guid;
    quoteId?: Guid;
    quote?: iQuote | null;
    title?: string;
    quantity?: number;
    unitPrice?: number;
    totalPrice?: number;
    totalValue?: number | null;
}

export interface iQuotePaginated {
    output: iQuote[];
    count: number;
}